ShortestPathLength/ShortestPathLength/Program.cs

```diff
@@ -12,20 +12,47 @@ static void Main(string[] args)
 
         public static int shortestPathLength(Node start, Node end)
         {
-            // Start at the start
-            // Create a set of nodes S that are reachable from start
-            // Create a set of nodes S' thar are reachable from S and have not yet been in any set
-            // Repeat until you reach end
-            // Return number of sets (+/-1?)
-
+            //create a variable to count steps
+            int steps = 0;
 
+            //make a scratch pad to keep track of which nodes have been visited and their respective values
+            Dictionary<Node, int> scratch = new Dictionary<Node, int>();
```

Comment:
What does "values" mean here?  Your usage is to save IDs.  Shouldn't this be tracking distance?